packages/angular/build/src/tools/vite/middlewares/chrome-devtools-middleware.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { randomUUID } from 'node:crypto';
+import { mkdirSync, readFileSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+import type { Connect } from 'vite';
+
+const CHROME_DEVTOOLS_ROUTE = '/.well-known/appspecific/com.chrome.devtools.json';
+
+export function createChromeDevtoolsMiddleware(
+  cacheDir: string,
+  projectRoot: string,
+): Connect.NextHandleFunction {
+  let devtoolsConfig: string;
+  const devtoolsConfigPath = join(cacheDir, 'com.chrome.devtools.json');
+
+  return function chromeDevtoolsMiddleware(req, res, next) {
+    if (req.url !== CHROME_DEVTOOLS_ROUTE) {
+      next();
+
+      return;
+    }
+
+    // We store the UUID and re-use it to ensure Chrome does not repeatedly ask for permissions when restarting the dev server.
+    try {
+      devtoolsConfig ??= readFileSync(devtoolsConfigPath, 'utf-8');
```

Comment:
Maybe I'm missing something, but what happens if you relocate the project folder? That doesn't seem to trigger deletion of the cached file.

Reply:
That is correct, we could add a check here to validate that the path is the same otherwise we could re-generate the entry.